tests/queries/0_stateless/02995_new_settings_history.sh

```diff
@@ -50,7 +50,7 @@ $CLICKHOUSE_LOCAL --query "
             )) AND (name NOT IN (
                 SELECT arrayJoin(tupleElement(changes, 'name'))
                 FROM system.settings_changes
-                WHERE type = 'MergeTree' AND splitByChar('.', version)[1]::UInt64 >= 24 AND splitByChar('.', version)[2]::UInt64 > 11
+                WHERE type = 'MergeTree' AND splitByChar('.', version)[1]::UInt64 >= 24 OR (splitByChar('.', version)[1]::UInt64 == 24 AND splitByChar('.', version)[2]::UInt64 > 11)
```

Comment:
I guess we want to check that
- the year is > 25 (*)
- OR the year is = 24 and the month is > 11.

(*) Right now, this first check seems too lax in your PR.